samples/query-features-with-arcade-expression/src/main/java/com/esri/arcgismaps/sample/queryfeatureswitharcadeexpression/components/QueryFeaturesWithArcadeExpressionViewModel.kt

```diff
@@ -0,0 +1,190 @@
+
+        Graphic(symbol = markerSymbol)
+    }
+
+    // data layer to be loaded from portal item
+    private var policeBeatsLayer: Layer? = null
+
+    // state flow to expose query results and status to UI
+    private val _queryStateFlow = MutableStateFlow(QueryState(loadState = LoadState.LOADING))
+    val queryStateFlow = _queryStateFlow.asStateFlow()
+
+    val graphicsOverlay = GraphicsOverlay()
+
+    // create a portal item with the itemId of the web map
+    val portal = Portal("https://www.arcgis.com/")
+    val portalItem = PortalItem(portal = portal, itemId = "539d93de54c7422f88f69bfac2aebf7d")
+
+    // create and add a map with with portal item
+    val arcGISMap = ArcGISMap(portalItem)
+
+    // create a map view proxy for handling interactions with the map view
+    val mapViewProxy = MapViewProxy()
+
+    // create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }
+
+            // get the RPD Beats layer from the map's operational layers
+            policeBeatsLayer = arcGISMap.operationalLayers.firstOrNull { layer ->
+                layer.id == "RPD_Reorg_9254"
+            }
+
+            // update query state, map is ready for user interaction
+            _queryStateFlow.value = QueryState(loadState = LoadState.READY_TO_START)
+        }
+
+        // add the marker graphic to the graphics overlay
+        graphicsOverlay.graphics.add(markerGraphic)
+    }
+
+    /**
+     * Handle a tap on the map view from the user
+     */
+    fun handleTap(point: Point, screenCoordinate: ScreenCoordinate) {
+        // update the marker location to where the user tapped on the map
+        markerGraphic.geometry = point
+        viewModelScope.launch {
+            // centre the viewpoint on where the user tapped on the map
+            mapViewProxy.setViewpointCenter(point)
+
+            // evaluate an Arcade expression on the tapped screen coordinate
+            evaluateArcadeExpression(screenCoordinate)
+        }
+    }
+
+    /**
+     * Evaluates an Arcade expression that returns crime in the last 60 days at the tapped
+     * [screenCoordinate] on the [arcGISMap] with the [policeBeatsLayer] and outputs the result
+     * to the [queryStateFlow] property.
+     */
+    private suspend fun evaluateArcadeExpression(screenCoordinate: ScreenCoordinate) {
+        policeBeatsLayer?.let { layer ->
+            // show the loading spinner as the Arcade evaluation can take time to complete
+            _queryStateFlow.value = QueryState(loadState = LoadState.LOADING)
+
+            // identify the layer and its elements based on the position tapped on the mapView and
+            // get the result
```

Comment:
I don't think this wording quite describes what this `identify` does. How about something like this?
> do an identify operation on the policeBeatsLayer, using the position tapped on the mapView, and get the result